akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -698,6 +701,24 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
     }
   }
 
+  // set the future that completes when the current stream for a given name completes
+  @tailrec
+  private def updateStreamCompletion(streamName: String, streamCompleted: Future[Done]): Unit = {
```

Comment:
if we don't want to use the `streamName` as key we could pass in the stream id to the restart method
```
  private val controlStreamId = 1
  private val ordinaryStreamId = 2
  private val largeStreamId = 3
```